go/vt/vttablet/tabletserver/throttle/base/self_metric_loadavg.go

```diff
@@ -55,13 +46,13 @@ func (m *LoadAvgSelfMetric) RequiresConn() bool {
 	return false
 }
 
-func (m *LoadAvgSelfMetric) Read(ctx context.Context, throttler ThrottlerMetricsPublisher, conn *connpool.Conn) *ThrottleMetric {
-	if runtime.GOOS != "linux" {
-		return loadavgOnlyAvailableOnLinuxMetric
-	}
+func (m *LoadAvgSelfMetric) Read(ctx context.Context, params *SelfMetricReadParams) *ThrottleMetric {
 	metric := &ThrottleMetric{
 		Scope: SelfScope,
 	}
+	if runtime.GOOS != "linux" {
+		return metric
+	}
```

Comment:
Previously, I threw an error. But that made unit testing + endtoend testing fail on MacOS machines, which we commonly use for development. 

I don't have a good story for MacOS. I could:
- Invoke `sysctl -n vm.loadavg`
- Invoke `uptime`
But I don't like invoking a command to get such info. Do you think we should?

Reply:
For `/proc/loadavg`:
```
The first three fields in this file are load average figures giving the number of jobs in the run queue (state R) or waiting for disk I/O (state D) averaged over 1, 5, and 15 minutes.
```

That seems to line up with `sysctl -n vm.loadavg`:
```
❯ uptime
15:39  up 10 days, 17:51, 5 users, load averages: 3.31 2.48 2.73
❯ sysctl -n vm.loadavg
{ 3.31 2.48 2.73 }
```

We don't have to support this, but if it's pretty easy it might be worth it -- largely for tests.

Reply:
Makes sense. Adding support for Mac!